pkgs/development/compilers/llvm/11/llvm-genode.patch

```diff
@@ -0,0 +1,157 @@
+diff --git a/cmake/modules/HandleLLVMOptions.cmake b/cmake/modules/HandleLLVMOptions.cmake
+index 5ef22eb493ba..f40625c7eed2 100644
+--- a/cmake/modules/HandleLLVMOptions.cmake
++++ b/cmake/modules/HandleLLVMOptions.cmake
+@@ -132,7 +132,7 @@ if(WIN32)
+     set(LLVM_ON_UNIX 0)
+   endif(CYGWIN)
+ else(WIN32)
+-  if(FUCHSIA OR UNIX)
++  if(FUCHSIA OR GENODE OR UNIX)
+     set(LLVM_ON_WIN32 0)
+     set(LLVM_ON_UNIX 1)
+     if(APPLE OR ${CMAKE_SYSTEM_NAME} MATCHES "AIX")
+@@ -140,9 +140,9 @@ else(WIN32)
+     else()
+       set(LLVM_HAVE_LINK_VERSION_SCRIPT 1)
+     endif()
+-  else(FUCHSIA OR UNIX)
++  else(FUCHSIA OR GENODE OR UNIX)
+     MESSAGE(SEND_ERROR "Unable to determine platform")
+-  endif(FUCHSIA OR UNIX)
++  endif(FUCHSIA OR GENODE OR UNIX)
+ endif(WIN32)
+ 
+ set(EXEEXT ${CMAKE_EXECUTABLE_SUFFIX})
+diff --git a/include/llvm/ADT/Triple.h b/include/llvm/ADT/Triple.h
+index 6bad18f19244..95459ca997c5 100644
+--- a/include/llvm/ADT/Triple.h
++++ b/include/llvm/ADT/Triple.h
+@@ -166,6 +166,7 @@ public:
+     DragonFly,
+     FreeBSD,
+     Fuchsia,
++    Genode,
+     IOS,
+     KFreeBSD,
+     Linux,
+@@ -502,6 +503,8 @@ public:
+ 
+   bool isOSDragonFly() const { return getOS() == Triple::DragonFly; }
+ 
++  bool isOSGenode() const { return getOS() == Triple::Genode; }
++
+   bool isOSSolaris() const {
+     return getOS() == Triple::Solaris;
+   }
+diff --git a/include/llvm/BinaryFormat/ELF.h b/include/llvm/BinaryFormat/ELF.h
+index bdcf10fd1640..46ed8e56cc31 100644
+--- a/include/llvm/BinaryFormat/ELF.h
++++ b/include/llvm/BinaryFormat/ELF.h
+@@ -348,6 +348,7 @@ enum {
+   ELFOSABI_AROS = 15,          // AROS
+   ELFOSABI_FENIXOS = 16,       // FenixOS
+   ELFOSABI_CLOUDABI = 17,      // Nuxi CloudABI
++  ELFOSABI_GENODE = 18,        // Genode
```

Comment:
Hm, this is a magic constant; what's its status? Is upstream considering adding it in 12 or 13 releases? Is it just a guess?

Reply:
This is what I should submit to upstream, but I've changed patch to set a temporary constant of 63.